gbin/gbin.py

```diff
@@ -143,19 +152,35 @@ def execute(self, args=None, always_exit=False, exit_if_failed=True, stdin=sys.s
         cmd.append(self._abs_path)
         if args:
             cmd = cmd + list(args)
-        if self.closest_venv:
-            process = self.closest_prepped_venv.run(cmd, always_exit=always_exit,
-                                                    exit_if_failed=exit_if_failed, stdin=stdin,
-                                                    stdout=stdout, stderr=stderr)
-        else:
-            process = subprocess.Popen(cmd, stdin=stdin, stdout=stdout, stderr=stderr)
-
-            exit_code = process.wait()
-            if always_exit:
-                sys.exit(exit_code)
-            if exit_if_failed and exit_code != 0:
-                sys.exit(exit_code)
-        return process
+        try:
+            if self.closest_venv:
+                process = self.closest_prepped_venv.run(cmd, always_exit=always_exit,
+                                                        exit_if_failed=exit_if_failed, stdin=stdin,
+                                                        stdout=stdout, stderr=stderr, env=env)
+            else:
+                env = env or os.environ
+                process = None
+                try:
+                    process = subprocess.Popen(cmd, stdin=stdin, stdout=stdout, stderr=stderr,
+                                               env=env)
+                    exit_code = process.wait()
+                except KeyboardInterrupt:
+                    if process:
+                        process.send_signal(signal.SIGINT)
+                        exit(process.wait())
+                    raise
```

Comment:
I know that this is equivalent to having the `raise` in the `else` block, but visually it would be cleaner to have an `else` block because it's more obvious what happens.